### PR TITLE
Add function to hook into external ACL function

### PR DESCRIPTION
**Pull request by skiptomyliu**
Add an external hook that allows us to integrate with an external ACL module.

### CODE REVIEW

File: confidant/routes/v1.py [hunk 1]
```diff
@@ -371,6 +372,14 @@ def get_credential_list():
 @app.route('/v1/credentials/<id>', methods=['GET'])
 @authnz.require_auth
 def get_credential(id):
+    if not acl_module_check(id):
```
Comment (ryan-lane) [inline confidant/routes/v1.py]: I think we need to pass in more info than just the ID. I think maybe `acl_module_check('get_credential', 'view', id)`

^^ that way we can make specific checks based on function, the permission within the function, and the resource it's being called against.

File: confidant/routes/v1.py [hunk 2]
```diff
@@ -102,13 +106,21 @@ def get_service(id):
     if authnz.user_is_user_type('service'):
         if not authnz.user_is_service(id):
             logging.warning('Authz failed for service {0}.'.format(id))
-            msg = 'Authenticated user is not authorized.'
+            msg = 'Service is not authorized.'
             return jsonify({'error': msg}), 401
     else:
+        logged_in_user = authnz.get_logged_in_user()
+        if not acl_module_check('get_service',
+                                metadata_only=metadata_only,
```
Comment (ryan-lane) [inline confidant/routes/v1.py]: I think we should check for allowed permissions here. So for instance, we'd pass in a list maybe? `['view', 'metadata']`

Comment (skiptomyliu) [reply confidant/routes/v1.py]: Just updated along with adding the `acl_module_check` to additional endpoints. 

Lemme know if this is what ya had in mind.  
```
metadata_only = request.args.get('metadata_only', default=False, type=bool)
...
acl_actions = ['metadata']
if not metadata_only:
    acl_actions.append('get')
```
In the current logic, users default will be able to view full credentials unless `metadata_only` explicitly set to `True`

File: confidant/routes/v1.py [hunk 3]
```diff
@@ -479,6 +533,16 @@ def get_credential_list():
 @app.route('/v1/credentials/<id>', methods=['GET'])
 @authnz.require_auth
 def get_credential(id):
+    if not acl_module_check('get_credential',
+                            actions=['get'],
+                            resource=id):
```
Comment (ryan-lane) [inline confidant/routes/v1.py]: We should followup in another PR to also support metadata here

File: confidant/settings.py [hunk 4]
```diff
@@ -491,3 +492,21 @@ def get(name, default=None):
     if encrypted_settings.registered(name):
         return encrypted_settings.get_secret(name)
     return globals().get(name, default)
+
+
+def load_module(module_path):
```
Comment (ryan-lane) [inline confidant/settings.py]: Can we move this to utils?

Comment (skiptomyliu) [reply confidant/settings.py]: moved to `util.misc`

**Pull request merged**